Solutions/Branwen/GUI.cs

```diff
@@ -9,324 +9,460 @@
-        /// <param name="e"></param>
-        private void buttonWipeDb_Click(object sender, EventArgs e)
-        {
-            buttonWipeDb.Enabled = false;
-            buttonSelectAndRunInventory.Enabled = false;
-            buttonSelectAndRunInventory.Text = "Working";
-            textBoxMediaDriveNumber.Enabled = false;
-            UseDBCheckBox.Enabled = false;
-
-            try
-            {
-                MySqlConnection mySqlConnection;
-                mySqlConnection = new MySqlConnection("server=box654.bluehost.com;user=lbkstud1_smedia;password=#sucK_my_d1ck;database=lbkstud1_SaurutobiMedia;");
-                mySqlConnection.Open();
-
-                string deleteStatement = "TRUNCATE TABLE SaurutobiMediaPaths; TRUNCATE TABLE SaurutobiMediaFiles;";
-                MySqlCommand deleteCommand = new MySqlCommand(deleteStatement, mySqlConnection);
-                deleteCommand.ExecuteNonQuery();
-            }
-            catch (Exception ex)
-            {
-                MessageBox.Show("Failed to Wipe:" + Environment.NewLine + ex.Message);
-            }
-            buttonWipeDb.Enabled = true;
-            buttonSelectAndRunInventory.Enabled = true;
-            buttonSelectAndRunInventory.Text = "Select Inventory Directory";
-            textBoxMediaDriveNumber.Enabled = true;
-            UseDBCheckBox.Enabled = true;
-        }
-    }
+	public partial class GUI : Form
+	{
+		private int fileCount;
+
+		public GUI()
+		{
+			InitializeComponent();
+		}
+
+		#region Button Handlers
+
+		/// <summary>
+		/// Button Handler, Selects Directory then does Inventory
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="e"></param>
+		private void SelectAndRunInventoryButton_Click(object sender, EventArgs e)
+		{
+			fileCount = 0;
+			try
+			{
+				//make this thing and all associated stuff go away
+				SelectAndRunInventoryButton.Enabled = false;
+				SelectAndRunInventoryButton.Text = "Working";
+				WipeDbButton.Enabled = false;
+				WipeDbButton.Text = "...";
+				MediaDriveNumberTextBox.Enabled = false;
+				UseDBCheckBox.Enabled = false;
+				ExportFileCheckBox.Enabled = false;
+				FolderBrowserDialog folderDialog = new FolderBrowserDialog();
+				folderDialog.Description = "Set Folder to Inventory";
+				if (folderDialog.ShowDialog() != DialogResult.OK)
+				{
+					SelectAndRunInventoryButton.Enabled = true;
+					SelectAndRunInventoryButton.Text = "Select Inventory Directory";
+					UseDBCheckBox.Enabled = true;
+					SetEnabledControls();
+					return;
+				}
+
+				DirectoryInfo[] topLevelDirectories = new DirectoryInfo(folderDialog.SelectedPath).GetDirectories();
+				string outputFile = Path.Combine(folderDialog.SelectedPath, "MediadriveInventory.xlsx");
+				if (outputFile == null || topLevelDirectories == null)
+				{
+					MessageBox.Show("This Directory is Invalid. Please try again");
+					return;
+				}
+
+				if (UseDBCheckBox.Checked == true)

```

Comment:
stupid me. I really wanted to be sure. spell it out for ya:)

On Thu, Dec 17, 2015 at 11:15 PM, Alex Ayerdi <notifications@github.com>
wrote:

> In Solutions/Branwen/GUI.cs
> <https://github.com/LbKStudios/Branwen/pull/5#discussion_r47999915>:
>
> > +					SelectAndRunInventoryButton.Enabled = true;
> > +					SelectAndRunInventoryButton.Text = "Select Inventory Directory";
> > +					UseDBCheckBox.Enabled = true;
> > +					SetEnabledControls();
> > +					return;
> > +				}
> > +
> > +				DirectoryInfo[] topLevelDirectories = new DirectoryInfo(folderDialog.SelectedPath).GetDirectories();
> > +				string outputFile = Path.Combine(folderDialog.SelectedPath, "MediadriveInventory.xlsx");
> > +				if (outputFile == null || topLevelDirectories == null)
> > +				{
> > +					MessageBox.Show("This Directory is Invalid. Please try again");
> > +					return;
> > +				}
> > +
> > +				if (UseDBCheckBox.Checked == true)
>
> Redundant. Checked already returns a boolean
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LbKStudios/Branwen/pull/5/files#r47999915>.
>